x/feeshare/ante/ante.go

```diff
@@ -62,29 +63,63 @@ type FeeSharePayoutEventOutput struct {
 	FeesPaid        sdk.Coins      `json:"fees_paid"`
 }
 
+func addNewFeeSharePayoutsForMsg(ctx sdk.Context, fsk FeeShareKeeper, toPay *[]sdk.AccAddress, m sdk.Msg) error {
+	if msg, ok := m.(*wasmtypes.MsgExecuteContract); ok {
+		contractAddr, err := sdk.AccAddressFromBech32(msg.Contract)
+		if err != nil {
+			return err
+		}
+
+		shareData, _ := fsk.GetFeeShare(ctx, contractAddr)
+
+		withdrawAddr := shareData.GetWithdrawerAddr()
+		if withdrawAddr != nil && !withdrawAddr.Empty() {
+			*toPay = append(*toPay, withdrawAddr)
+		}
+	}
+
+	return nil
+}
+
 // FeeSharePayout takes the total fees and redistributes 50% (or param set) to the contract developers
 // provided they opted-in to payments.
-func FeeSharePayout(ctx sdk.Context, bankKeeper BankKeeper, totalFees sdk.Coins, revKeeper FeeShareKeeper, msgs []sdk.Msg) error {
-	params := revKeeper.GetParams(ctx)
+func FeeSharePayout(ctx sdk.Context, bankKeeper BankKeeper, totalFees sdk.Coins, fsk FeeShareKeeper, msgs []sdk.Msg) error {
+	params := fsk.GetParams(ctx)
 	if !params.EnableFeeShare {
 		return nil
 	}
 
 	// Get valid withdraw addresses from contracts
 	toPay := make([]sdk.AccAddress, 0)
-	for _, msg := range msgs {
-		if _, ok := msg.(*wasmtypes.MsgExecuteContract); ok {
-			contractAddr, err := sdk.AccAddressFromBech32(msg.(*wasmtypes.MsgExecuteContract).Contract)
+
+	// validAuthz checks if the msg is an authz exec msg and if so, call the validExecuteMsg for each
+	// inner msg. If it is a CosmWasm execute message, that logic runs for nested functions.
+	validAuthz := func(execMsg *authz.MsgExec) error {
+		for _, v := range execMsg.Msgs {
+			var innerMsg sdk.Msg
+			if err := json.Unmarshal(v.Value, &innerMsg); err != nil {
```

Comment:
I think this is always going to fail. You should use SDK codec to correctly unmarshal any type to an interface.

Reply:
To add on to what aleem is saying, It turns out authz.MsgExec has a built in function called "GetMessages()" which returns an arrray of sdk.Msg. Way easier than looping through and trying to unmarshal.